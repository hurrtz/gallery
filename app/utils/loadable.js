import React, { lazy, Suspense } from 'react';

const loadable = (
  importFunc,
  /* istanbul ignore next: jest bug ignoring tests with matching default parameter */
  { fallback = null } = { fallback: null },
) => {
  const LazyComponent = lazy(importFunc);

  return props => (
    <Suspense fallback={fallback}>
      <LazyComponent {...props} />
    </Suspense>
  );
};

export default loadable;
