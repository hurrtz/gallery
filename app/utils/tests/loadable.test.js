import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import loadable from '../loadable';

const renderer = new ShallowRenderer();
const content = () => import('./mockComponent');

describe('<Loadable />', () => {
  it('should render and match the snapshot', () => {
    const Loadable = loadable(content);

    renderer.render(<Loadable />);

    const renderedOutput = renderer.getRenderOutput();

    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render without a given fallback component and match the snapshot', () => {
    const Loadable = loadable(content, {
      fallback: false,
    });

    renderer.render(<Loadable />);

    const renderedOutput = renderer.getRenderOutput();

    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render with a given fallback component and match the snapshot', () => {
    const Loadable = loadable(content, {
      fallback: <div>fallback</div>,
    });

    renderer.render(<Loadable />);

    const renderedOutput = renderer.getRenderOutput();

    expect(renderedOutput).toMatchSnapshot();
  });
});
