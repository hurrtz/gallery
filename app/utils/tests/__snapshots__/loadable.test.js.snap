// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loadable /> should render and match the snapshot 1`] = `
<React.Suspense
  fallback={null}
>
  <UNDEFINED />
</React.Suspense>
`;

exports[`<Loadable /> should render with a given fallback component and match the snapshot 1`] = `
<React.Suspense
  fallback={
    <div>
      fallback
    </div>
  }
>
  <UNDEFINED />
</React.Suspense>
`;

exports[`<Loadable /> should render without a given fallback component and match the snapshot 1`] = `
<React.Suspense
  fallback={false}
>
  <UNDEFINED />
</React.Suspense>
`;
